median/plugin.go

```diff
@@ -25,6 +25,20 @@ func (p *Plugin) NewMedianFactory(ctx context.Context, provider types.MedianProv
 	var ctxVals loop.ContextValues
 	ctxVals.SetValues(ctx)
 	lggr := logger.With(p.Logger, ctxVals.Args()...)
+
+	var reportCodec median.ReportCodec
```

Comment:
WDYT about moving this block after the `factory` declaration, so that you can just set `factory.ReportCodec` directly, instead of having this intermediate var?
```suggestion
```